core/src/data_sources/data_source.rs

```diff
@@ -130,6 +133,64 @@ pub struct DataSource {
     config: DataSourceConfig,
 }
 
+fn target_document_tokens_offsets(
+    offsets: Vec<usize>,
+    space: usize,
```

Comment:
let's call it `tokens_to_grow` maybe?

I would even go as far as suggesting operating at the chunk level here (chunks_to_grow) and have a simple `(target_document_tokens - chunks * max_chunk_size) / max_chunk_size`

but that's not a big deal

Reply:
For writing tests, using chunks_to_grow will make the tests easier to write and understand I think :+1:

Reply:
You mean i should change the logic so the argument is how many chunks can be added?